### PR TITLE
IntelliJ Project Starter

### LINKED ISSUE #6: IntelliJ Project starter
- SpringBoot
- JPA
- JUnit
- PostgresSQL Driver
- Spring Testing

### PR DESCRIPTION
**Pull request by dershao**
Project setup

Closes #6

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-jpa</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-rest</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>com.h2database</groupId>
+            <artifactId>h2</artifactId>
```
Comment (mpfan) [inline pom.xml:37]: Should use Postgres instead of h2.

**Review by mpfan — CHANGES_REQUESTED** [review]

File: src/main/resources/application.properties [hunk 2]
```diff
@@ -0,0 +1,7 @@
+spring.datasource.driver-class-name=org.postgresql.Driver
+spring.datasource.url= jdbc:postgresql://${DB_HOST}:${DB_PORT}/${DB_NAME}
+spring.datasource.username=${DB_USER}
+spring.datasource.password=${DB_PASS}
+spring.jpa.hibernate.ddl-auto=create-drop
+spring.jpa.database-platform=postgres
+spring.jpa.properties.hibernate.dialect=org.hibernate.dialect.PostgreSQLDialect
```
Comment (mpfan) [inline src/main/resources/application.properties:7]: Note for later: remember to configure the db variables.
I think heroku has an option for hosting a postgres db (along with a web app). If not we would need to host the db somewhere.

**Review by mpfan — APPROVED** [review]

**Comment by Weebian** [comment]
Looks good

**Comment by Souheil-Yazji** [comment]
LGM

**Review by Weebian — APPROVED** [review]
looks good

File: pom.xml [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+    <version>1.0-SNAPSHOT</version>
+
+    <dependencies>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-web</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-devtools</artifactId>
```
Comment (Souheil-Yazji) [inline pom.xml:23]: needed?

File: pom.xml [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+            <optional>true</optional>
+        </dependency>
+        <dependency>
+            <groupId>org.springframework.boot</groupId>
+            <artifactId>spring-boot-starter-data-rest</artifactId>
+        </dependency>
+        <dependency>
+            <groupId>org.postgresql</groupId>
+            <artifactId>postgresql</artifactId>
+            <version>42.2.19</version>
```
Comment (Souheil-Yazji) [inline pom.xml:34]: might cause issues with heroku deploy support

File: src/main/resources/application.properties (hunk 2 above)
Comment (dershao) [reply src/main/resources/application.properties:7]: Heroku does fortunately - that's actually what we'll be doing in this week's lab

File: pom.xml (hunk 3 above)
Comment (dershao) [reply pom.xml:23]: Not necessary no - its for live reloads

File: src/main/resources/application.properties (hunk 2 above)
Comment (Souheil-Yazji) [reply src/main/resources/application.properties:7]: heroku provides support for postgreSQL, see 
https://dev.to/prisma/how-to-setup-a-free-postgresql-database-on-heroku-1dc1

File: pom.xml (hunk 3 above)
Comment (mpfan) [reply pom.xml:23]: Live reloads would be nice.

File: src/main/resources/application.properties (hunk 2 above)
Comment (mpfan) [reply src/main/resources/application.properties:7]: Refer to #9

**Pull request merged**